### PR TITLE
[optimization] Remove camera dependency

### PR DESCRIPTION
**Pull request by troublemaker52025**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [x] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by troublemaker52025** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

**Comment by minggo** [comment]
>  list all changes, affected features and the scope of violation.

 Should describe it.

**Comment by troublemaker52025** [comment]
# introduces breaking changes

affected features and the scope of violation

##  for js engine

* Change the way the split function is called, from a static member function to a member function, user scripts will not be able to use the static method directly.
* Change the form parameter of split function, user scripts need to change the parameter passed to it.
* Allow users to call functions of the same name in the native engine via scripts.
* Remove the dependency on the Camera class

## for native engine

* Allow users to call functions of the same name in the native engine via scripts.
* Change the formal arguments of the split function to be consistent with the js engine parameters.
* Remove the dependency on the scene::Camera class

File: cocos/core/geometry/frustum.ts [hunk 1]
```diff
@@ -125,21 +124,20 @@ export class Frustum {
     /**
      * @en Calculate the splitted frustum.
      * @zh 创建一个新的截锥体。
```
Comment (minggo) [inline cocos/core/geometry/frustum.ts]: It doesn't create a new frustum now. The comment should be changed too.

Comment (troublemaker52025) [reply cocos/core/geometry/frustum.ts]: sure

**Review by jk20012001 — APPROVED** [review]

**Pull request merged**